src/components/select/filterable-select/filterable-select.spec.tsx

```diff
@@ -613,6 +617,204 @@ describe("FilterableSelect", () => {
     });
   });
 
+  describe("when the filter text contains whitespace,", () => {
```

Comment:
comment: I think it would be better to avoid over nesting the describe blocks here. Obviously there's already a lot of nesting in the spec files for the Select components but I think we should avoid adding more. The console output will also be a little confusing as right now it will output like so:

```
when the filter text contains whitespace
  which is leading whitespace:
     the input value is trimmed and filled correctly

Reply:
Maybe something like this would work:

```
it("should render the input with value trimmed and filled correctly when the filterText contains leading whitespace", () => {
...
it("should render the input with selection range set correctly when the filterText contains leading whitespace", () => {

Reply:
Happy to chat about this on slack tomorrow, also don't mind one `describe` if you prefer to encapsulate it a bit more than the above would, but I think the console output needs tidying up and some nesting removed

Reply:
**suggestion(non-blocking)**: what do you think about the following structure so we could remove the top `describe` block? ("when the filter text contains whitespace,")

```
describe("when filter text contains leading whitespace", () => { /* tests... */});
describe("when filter text contains whitespace that is neither leading or trailing", () => { /* tests... */});
describe("when filter text is only whitespace", () => { /* tests... */});
describe("when filter text contains trailing whitespace", () => { /* tests... */});
describe("when filter text contains both leading and trailing whitespace", () => { /* tests... */});
```